### DISCUSSION TITLE
Code search OR condition not working as expected

### DISCUSSION THREAD

**Discussion by mikechari** [community/community#125409] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi, this may be asked before but i didn't found anything related, here is what is happening:

I want to do a search for code occurrences containing for example **textA OR textB** here is my request
`https://api.github.com/search/code?q=stringA OR stringB+repo:repo`

The case:
I know that **textA** is in two files and **textB** is only on 1 file
If i search for **textA** i get **2** occurrences, if i search for **textA OR textB** i get only **1** occurrences (in the occurrence the textA is also in the same file as textB).

Why i am getting only 1 if it is an OR condition it should be 2? I am doing something wrong?

The main intention of the search is to find all files that contain one text or the other.

**Comment by look** [comment, marked as answer]
@mikechari sorry for the confusion, but the search API still uses the [old GitHub code search syntax](https://docs.github.com/en/search-github/searching-on-github/searching-code) which doesn't support Boolean operators. We plan to release a new API with the new search syntax, though unfortunately I can't say when that would be publicly available. For now, the only way to do what you want is through the web UI.
